test/functional/wallet_basic.py

```diff
@@ -336,6 +351,25 @@ def run_test(self):
         # This will raise an exception for importing an invalid pubkey
         assert_raises_rpc_error(-5, "Pubkey is not a valid public key", self.nodes[0].importpubkey, "5361746f736869204e616b616d6f746f")
 
+        # send with explicit fee
+        self.sync_all([self.nodes[0:3]])
+        self.log.info("test explicit fee (sendtoaddress)")
+        self.nodes[0].generate(1)
+        prebalance = self.nodes[2].getbalance()
+        assert prebalance > 2
+        txid = self.nodes[2].sendtoaddress(
+            address=self.nodes[1].getnewaddress(),
+            amount=1.0,
+            conf_target=0.00002500,
+            estimate_mode='EXPLICIT',
+        )
+        tx_size = self.get_vsize(self.nodes[2].gettransaction(txid)['hex'])
+        self.sync_all([self.nodes[0:3]])
+        self.nodes[0].generate(1)
```

Comment:
Since the block is mined by node 0, and the balance is on node 2, you need another `sync_all` here or else there's a race condition.

Reply:
@luke-jr Thanks, fixed!